spark/unitycatalog/src/test/java/io/sparkuctest/UCDeltaTableDMLTest.java

```diff
@@ -0,0 +1,276 @@
+    withNewTable("merge_update_test", "id INT, value STRING", tableType, tableName -> {
+      // Setup target table
+      sql("INSERT INTO %s VALUES (1, 'old1'), (2, 'old2'), (3, 'old3')", tableName);
+
+      // Perform merge to update existing records
+      withNewTable("merge_update_source", "id INT, value STRING", tableType, sourceTable -> {
+        sql("INSERT INTO %s VALUES (2, 'updated2'), (3, 'updated3')", sourceTable);
+
+        sql("MERGE INTO %s AS target " +
+            "USING %s AS source " +
+            "ON target.id = source.id " +
+            "WHEN MATCHED THEN UPDATE SET value = source.value", tableName, sourceTable);
+      });
+
+      // Verify merge result
+      check(tableName, List.of(
+          List.of("1", "old1"),
+          List.of("2", "updated2"),
+          List.of("3", "updated3")
+      ));
+    });
+  }
+
+  @ParameterizedTest
+  @MethodSource("allTableTypes")
+  public void testMergeCombinedInsertAndUpdate(TableType tableType) throws Exception {
+    withNewTable("merge_combined_test", "id INT, name STRING, status STRING", tableType, tableName -> {
+      // Setup target table
+      sql("INSERT INTO %s VALUES (1, 'Alice', 'active'), (2, 'Bob', 'inactive')", tableName);
+
+      // Perform merge with both insert and update
+      withNewTable("merge_combined_source", "id INT, name STRING, status STRING", tableType, sourceTable -> {
+        sql("INSERT INTO %s VALUES " +
+            "(2, 'Bob', 'active'), (3, 'Charlie', 'active'), (4, 'Diana', 'pending')", sourceTable);
+
+        sql("MERGE INTO %s AS target " +
+            "USING %s AS source " +
+            "ON target.id = source.id " +
+            "WHEN MATCHED THEN UPDATE SET status = source.status " +
+            "WHEN NOT MATCHED THEN INSERT (id, name, status) VALUES (source.id, source.name, source.status)", tableName, sourceTable);
+      });
+
+      // Verify merge result
+      check(tableName, List.of(
+          List.of("1", "Alice", "active"),
+          List.of("2", "Bob", "active"),
+          List.of("3", "Charlie", "active"),
+          List.of("4", "Diana", "pending")
+      ));
+    });
+  }
+
+  @ParameterizedTest
+  @MethodSource("allTableTypes")
+  public void testMergeWithDeleteAction(TableType tableType) throws Exception {
+    withNewTable("merge_delete_test", "id INT, active BOOLEAN", tableType, tableName -> {
+      // Setup target table
+      sql("INSERT INTO %s VALUES (1, true), (2, true), (3, false), (4, true)", tableName);
+
+      // Perform merge with delete action
+      withNewTable("merge_delete_source", "id INT, active BOOLEAN", tableType, sourceTable -> {
+        sql("INSERT INTO %s VALUES (2, false), (3, false), (5, true)", sourceTable);
+
+        sql("MERGE INTO %s AS target " +
+            "USING %s AS source " +
+            "ON target.id = source.id " +
+            "WHEN MATCHED AND source.active = false THEN DELETE " +
+            "WHEN MATCHED THEN UPDATE SET active = source.active " +
```

Comment:
Does this condition have real meaning ? because from the above target table records and source table records, there is actully no record hitting this condition, right ?  maybe we can remove this line directly ?